core/frontend/src/tile/ImdlReader.ts

```diff
@@ -246,10 +244,26 @@ export class ImdlReader extends GltfReader {
     const cacheable = !isGlyph && !isTileSection;
     const params = new RenderTexture.Params(cacheable ? name : undefined, textureType);
 
-    const bytes = this._binaryData.subarray(byteOffset, byteOffset + byteLength);
-    const format = namedTex.format;
-    const imageSource = new ImageSource(bytes, format);
-    return this._system.createTextureFromImageSource(imageSource, this._iModel, params);
+    const bufferViewId = JsonUtils.asString(namedTex.bufferView);
+    const bufferViewJson = 0 !== bufferViewId.length ? this._bufferViews[bufferViewId] : undefined;
+
+    const useExternalTexture = Id64.isValidId64(name) && IModelApp.tileAdmin.enableExternalTextures && undefined === bufferViewJson;

```

Comment:
Is there nothing in the tile content itself that indicates whether the texture is embedded or should be requested from backend? Seems wrong to rely on `IModelApp` setting.

Reply:
I think we could rely on the presence of 'bufferViewJson' - I'll look at reorganizing that logic on the frontend so it's potentially simpler.

Reply:
Greatly simplified this code.